### PR TITLE
Handle login exception when logging out

### PR DESCRIPTION
**Pull request by sancar**
Made logout last so that all steps are done even when an exception
thrown form logout method.

Tests are on the enterprise repo.
https://github.com/hazelcast/hazelcast-enterprise/pull/3493
related to https://github.com/hazelcast/hazelcast/issues/16482

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/client/impl/ClientEndpointImpl.java [hunk 1]
```diff
@@ -215,19 +215,22 @@ public void destroy() throws LoginException {
         destroyed = true;
         nodeEngine.onClientDisconnected(getUuid());
```
Comment (kwart) [inline hazelcast/src/main/java/com/hazelcast/client/impl/ClientEndpointImpl.java]: try-catch?

File: hazelcast/src/main/java/com/hazelcast/client/impl/ClientEndpointImpl.java [hunk 2]
```diff
@@ -215,19 +215,22 @@ public void destroy() throws LoginException {
...
 
-        LoginContext lc = loginContext;
-        if (lc != null) {
-            lc.logout();
-        }
-
         for (UUID registrationId : removeListenerActions.keySet()) {
             removeAndCallRemoveAction(registrationId);
```
Comment (kwart) [inline hazelcast/src/main/java/com/hazelcast/client/impl/ClientEndpointImpl.java]: try-catch?

File: hazelcast/src/main/java/com/hazelcast/client/impl/ClientEndpointImpl.java [hunk 3]
```diff
@@ -215,19 +215,22 @@ public void destroy() throws LoginException {
...
         }
 
         for (UUID txnId : transactionContextMap.keySet()) {
             removedAndRollbackTransactionContext(txnId);
```
Comment (kwart) [inline hazelcast/src/main/java/com/hazelcast/client/impl/ClientEndpointImpl.java]: try-catch?

**Comment by sancar** [comment]
@kwart The exceptions are already handled in each method. There is no need for extra try catch.

**Pull request merged**